app/src/main/java/com/epfl/drawyourpath/mainpage/fragments/helperClasses/MessagesAdapter.kt

```diff
@@ -71,7 +74,25 @@ class MessagesAdapter(private val messages: List<Message>, private val userId: S
             }
             VIEW_TYPE_RUN_PATH -> {
                 val runPathHolder = holder as RunPathMessageViewHolder
-                // Customization for RunPath message content can be added here
+                val content = message.content as MessageContent.RunPath
+
+                val run = content.run
+
+                val runCoordinates: List<LatLng> = run.getPath().getPoints().flatten()
+                val apiKey = "AIzaSyCE8covSYZE_sOv4Z-HaoljRlNOTV8cKRk"
+
+                val staticMapUrl = Utils.getStaticMapUrl(runCoordinates, apiKey)
+                // TODO: will be refactor with the creation of a bitmap directly with the run coordinates
+                Glide.with(holder.itemView.context)
+                    .load(staticMapUrl)
+                    .placeholder(R.drawable.map_loading_placeholder)
+                    .into(runPathHolder.mapImageView)
+
+                runPathHolder.dateTextView.text = run.getDate()

```

Comment:
This code with Glider is also present in `RunAdapter`, it should be refactored. Maybe a function to set the bitmap image of an `ImageView` from a list of list of points.

Reply:
Not sure what you mean. Is there another way of getting a bitmap image of a map with the path?

Reply:
No, it's not about the logic, just that the exact same code with the api key is also in `RunAdapter`.

Reply:
Ok, understood. I created a Utils function for it and refactored the code in there and the RunAdapter to call it.

Reply:
Perfect, but there was some changes in the main branch to set a local representation of the image while it loads:
```kotlin
imageView.setImageBitmap(Utils.coordinatesToBitmap(run.getPath().getPoints()))
```
It should also go in the new function.

Reply:
I see, I added that.